### PR TITLE
Update dotnet sample with Entra ID

### PR DESCRIPTION
**Pull request by CawaMS**
## Purpose

* ...

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Comment by CawaMS** [comment]
@philon-msft @yugangw-msft can one of you please review and approve the PR? This is for Entra ID update

File: quickstart/dotnet/README.md [hunk 1]
```diff
@@ -19,22 +19,41 @@ This sample shows you how to incorporate Azure Cache for Redis into a C# .NET Fr
 - Azure Cache for Redis cache - [create one](https://docs.microsoft.com/azure/azure-cache-for-redis/quickstart-create-redis)
 - [Visual Studio 2019](https://www.visualstudio.com/downloads/)
 - [.NET Framework 4 or higher](https://www.microsoft.com/net/download/dotnet-framework-runtime), which is required by the StackExchange.Redis client.
+- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
 
 ## Set up the working environment
 
+### 1. Set up local credential for using Entra ID
+This sample uses Microsoft Entra ID for connecting to an Azure Cache for Redis instance.
+The following line of code in *ContosoTeamStats/RedisConnection.cs* obtains the default credential from your local machine or an Azure resource as the identity for authentication and authorization.
```
Comment (yugangw-msft) [inline quickstart/dotnet/README.md:28]: `Redistest/RedisConnections.cs`?

File: quickstart/dotnet/README.md [hunk 2]
```diff
@@ -19,22 +19,41 @@ This sample shows you how to incorporate Azure Cache for Redis into a C# .NET Fr
...
+
+```cli
+az login
+```
+
+For other methods of sign into Azure with Azure CLI, such as using a Service Principal, see [Sign into Azure with Azure CLI](https://learn.microsoft.com/cli/azure/authenticate-azure-cli)
+
+### 2. Point to an Azure Cache for Redis instance from local configuration
+
 Create a file on your computer named *CacheSecrets.config* and place it in a location where it won't be checked in with the source code of your sample application. For this quickstart, the *CacheSecrets.config* file is located here, *C:\AppSecrets\CacheSecrets.config*.
```
Comment (yugangw-msft) [inline quickstart/dotnet/README.md:44]: Do we still need the secret?

File: quickstart/dotnet/Redistest/Redistest.csproj [hunk 3]
```diff
@@ -52,11 +53,11 @@
     <None Include="App.config" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="StackExchange.Redis">
-      <Version>2.*</Version>
+    <PackageReference Include="Microsoft.Azure.StackExchangeRedis">
+      <Version>3.1.0</Version>
```
Comment (yugangw-msft) [inline quickstart/dotnet/Redistest/Redistest.csproj:57]: Shal we use `3.*` to pull in the latest update w/o breaking change?

File: quickstart/dotnet/README.md [hunk 4]
```diff
@@ -19,22 +19,41 @@ This sample shows you how to incorporate Azure Cache for Redis into a C# .NET Fr
 - Azure Cache for Redis cache - [create one](https://docs.microsoft.com/azure/azure-cache-for-redis/quickstart-create-redis)
 - [Visual Studio 2019](https://www.visualstudio.com/downloads/)
 - [.NET Framework 4 or higher](https://www.microsoft.com/net/download/dotnet-framework-runtime), which is required by the StackExchange.Redis client.
+- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
 
 ## Set up the working environment
 
+### 1. Set up local credential for using Entra ID
+This sample uses Microsoft Entra ID for connecting to an Azure Cache for Redis instance.
```
Comment (philon-msft) [inline quickstart/dotnet/README.md:27]: "uses Microsoft Entra ID to authenticate connections to an Azure Cache for Redis resource"

File: quickstart/dotnet/README.md (hunk 1 above)
Comment (philon-msft) [inline quickstart/dotnet/README.md:28]: Link to docs on how `DefaultAzureCredential` works

File: quickstart/dotnet/README.md [hunk 5]
```diff
@@ -19,22 +19,41 @@ This sample shows you how to incorporate Azure Cache for Redis into a C# .NET Fr
...
+### 2. Point to an Azure Cache for Redis instance from local configuration
+
 Create a file on your computer named *CacheSecrets.config* and place it in a location where it won't be checked in with the source code of your sample application. For this quickstart, the *CacheSecrets.config* file is located here, *C:\AppSecrets\CacheSecrets.config*.
 
 Edit the *CacheSecrets.config* file and add the following contents:
 
 ```xml
 <appSettings>
-    <add key="CacheConnection" value="<cache-name>.redis.cache.windows.net,abortConnect=false,ssl=true,allowAdmin=true,password=<access-key>"/>
+    <add key="RedisCacheName" value="<cache-name>.redis.cache.windows.net"/>
```
Comment (philon-msft) [inline quickstart/dotnet/README.md:50]: "RedisCacheHostName"?

Comment (philon-msft) [reply quickstart/dotnet/README.md:50]: actually it looks like the code loads a setting named ""RedisHostName"

File: quickstart/dotnet/README.md [hunk 6]
```diff
@@ -19,22 +19,41 @@ This sample shows you how to incorporate Azure Cache for Redis into a C# .NET Fr
 - Azure Cache for Redis cache - [create one](https://docs.microsoft.com/azure/azure-cache-for-redis/quickstart-create-redis)
 - [Visual Studio 2019](https://www.visualstudio.com/downloads/)
 - [.NET Framework 4 or higher](https://www.microsoft.com/net/download/dotnet-framework-runtime), which is required by the StackExchange.Redis client.
+- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
```
Comment (stanleysmall-microsoft) [inline quickstart/dotnet/README.md:22]: Remove `en-us` to automatically select a langauge

File: quickstart/dotnet/README.md (hunk 1 above)
Comment (stanleysmall-microsoft) [inline quickstart/dotnet/README.md:28]: Nit: use backticks `ContosoTeamStats/RedisConnection.cs` instead of italics *ContosoTeamStats/RedisConnection.cs*

Comment (CawaMS) [reply quickstart/dotnet/README.md:28]: fixed.

File: quickstart/dotnet/README.md (hunk 2 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:44]: fixed

File: quickstart/dotnet/Redistest/Redistest.csproj (hunk 3 above)
Comment (CawaMS) [reply quickstart/dotnet/Redistest/Redistest.csproj:57]: fixed

File: quickstart/dotnet/README.md (hunk 4 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:27]: fixed.

File: quickstart/dotnet/README.md (hunk 1 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:28]: fixed

File: quickstart/dotnet/README.md (hunk 5 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:50]: fixed

File: quickstart/dotnet/README.md (hunk 6 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:22]: fixed

File: quickstart/dotnet/README.md (hunk 1 above)
Comment (CawaMS) [reply quickstart/dotnet/README.md:28]: fixed

**Comment by CawaMS** [comment]
@yugangw-msft  @philon-msft @stanleysmall-microsoft can one of you please approve? I fixed all the comments you had.

**Review by stanleysmall-microsoft — APPROVED** [review]

**Comment by CawaMS** [comment]
@yugangw-msft @philon-msft fixed everything. please approve.

**Pull request merged**